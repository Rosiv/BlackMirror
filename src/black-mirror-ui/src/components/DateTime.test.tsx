import * as React from 'react';
import renderer from 'react-test-renderer';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
const customMuiTheme = getMuiTheme(darkBaseTheme);
import DateTime from './DateTime';

it('renders and is compatible with the snapshot - text not provided', () => {
    const tree = renderer
        .create(
        <MuiThemeProvider muiTheme={customMuiTheme}>
            <DateTime
                value={'012345'}
                nestedLevel={0}
                text={''}
            />
        </MuiThemeProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders and is compatible with the snapshot - text provided', () => {
    const tree = renderer
        .create(
        <MuiThemeProvider muiTheme={customMuiTheme}>
            <DateTime
                value={'012345'}
                nestedLevel={0}
                text={'This is a test.'}
            />
        </MuiThemeProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
