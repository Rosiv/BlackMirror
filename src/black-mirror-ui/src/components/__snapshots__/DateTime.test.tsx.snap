// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and is compatible with the snapshot - text not provided 1`] = `
<div>
  <span
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(255, 255, 255, 1)",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "marginLeft": 0,
            "muiPrepared": true,
            "paddingBottom": 16,
            "paddingLeft": 72,
            "paddingRight": 16,
            "paddingTop": 20,
            "position": "relative",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "display": "block",
              "fill": "#fff59d",
              "height": 24,
              "left": 4,
              "margin": 12,
              "muiPrepared": true,
              "position": "absolute",
              "top": 12,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
        <div
          style={
            Object {
              "muiPrepared": true,
            }
          }
        >
          012345
        </div>
        <div
          style={
            Object {
              "WebkitBoxOrient": null,
              "WebkitLineClamp": null,
              "color": "rgba(255, 255, 255, 0.7)",
              "display": null,
              "fontSize": 14,
              "height": 16,
              "lineHeight": "16px",
              "margin": 0,
              "marginTop": 4,
              "muiPrepared": true,
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          Creation time
        </div>
      </div>
    </div>
  </span>
</div>
`;

exports[`renders and is compatible with the snapshot - text provided 1`] = `
<div>
  <span
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(255, 255, 255, 1)",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "marginLeft": 0,
            "muiPrepared": true,
            "paddingBottom": 16,
            "paddingLeft": 72,
            "paddingRight": 16,
            "paddingTop": 20,
            "position": "relative",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(255, 255, 255, 1)",
              "display": "block",
              "fill": "#fff59d",
              "height": 24,
              "left": 4,
              "margin": 12,
              "muiPrepared": true,
              "position": "absolute",
              "top": 12,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
        <div
          style={
            Object {
              "muiPrepared": true,
            }
          }
        >
          012345
        </div>
        <div
          style={
            Object {
              "WebkitBoxOrient": null,
              "WebkitLineClamp": null,
              "color": "rgba(255, 255, 255, 0.7)",
              "display": null,
              "fontSize": 14,
              "height": 16,
              "lineHeight": "16px",
              "margin": 0,
              "marginTop": 4,
              "muiPrepared": true,
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          This is a test.
        </div>
      </div>
    </div>
  </span>
</div>
`;
